diff -Naur imagemagick.js imagemagick.js
--- imagemagick.js	2018-01-20 13:24:12.000000000 -0600
+++ imagemagick.js	2018-01-20 13:24:21.000000000 -0600
@@ -153,13 +153,20 @@
         result = stdout;
       } else {
         result = parseIdentify(stdout);
-        geometry = result['geometry'].split(/x/);
+        if (result.geometry) {
+          geometry = result['geometry'].split(/x/);
 
-        result.format = result.format.match(/\S*/)[0]
-        result.width = parseInt(geometry[0]);
-        result.height = parseInt(geometry[1]);
-        result.depth = parseInt(result.depth);
-        if (result.quality !== undefined) result.quality = parseInt(result.quality) / 100;
+          result.format = result.format.match(/\S*/)[0]
+          result.width = parseInt(geometry[0]);
+          result.height = parseInt(geometry[1]);
+          result.depth = parseInt(result.depth);
+          if (result.quality !== undefined) {
+            result.quality = parseInt(result.quality) / 100;
+          }
+        } else {
+          result.width = 0;
+          result.height = 0;
+        }
       }
     }
     callback(err, result);
